ports/views.py

```diff
@@ -0,0 +1,293 @@
+                      'letter': letter.upper(),
+                      'portscount': portscount
+                  })
+
+def variantlist(request, variant):
+    all_objects = Variant.objects.filter(variant=variant)
+    all_objects_count = all_objects.count()
+    paginated_objects = Paginator(all_objects, 100)
+    page = request.GET.get('page', 1)
+    try:
+        objects = paginated_objects.get_page(page)
+    except PageNotAnInteger:
+        objects = paginated_objects.get_page(1)
+    except EmptyPage:
+        objects = paginated_objects.get_page(paginated_objects.num_pages)
+    return render(request, 'ports/variantlist.html', {
+        'objects': objects,
+        'variant': variant,
+        'all_objects_count': all_objects_count
+    })
+
+
+def portdetail(request, name):
+    port = Port.objects.get(name=name)
+    id = port.id
+    maintainers = Maintainer.objects.filter(ports__name=name)
+    dependencies = Dependency.objects.filter(port_name_id=id)
+    variants = Variant.objects.filter(port_id=id)
+
+    builders = Builder.objects.values_list('name', flat=True)
+    build_history = {}
+    for builder in builders:
+        build_history[builder] = BuildHistory.objects.filter(builder_name__name=builder, port_name=name).order_by('-time_start')
+    return render(request, 'ports/portdetail.html', {
+        'port': port,
+        'build_history': build_history,
+        'maintainers': maintainers,
+        'dependencies': dependencies,
+        'variants': variants,
+        'builders_list': builders,
+    })
+
+
+def all_builds_view(request):
+    filter_applied = False
+    if request.method == 'POST':
+        if request.POST['status-filter']:
+            filter_by = request.POST['status-filter']
+            if filter_by == "All Builds":
+                all_builds = BuildHistory.objects.all().order_by('-time_start')
+            else:
+                all_builds = BuildHistory.objects.filter(status=filter_by).order_by('-time_start')
+                filter_applied = filter_by
+        else:
+            return HttpResponse("Something went wrong")
+    else:
+        all_builds = BuildHistory.objects.all().order_by('-time_start')
+    paginated_builds = Paginator(all_builds, 100)
+    page = request.GET.get('page', 1)
+    try:
+        builds = paginated_builds.get_page(page)
+    except PageNotAnInteger:
+        builds = paginated_builds.get_page(1)
+    except EmptyPage:
+        builds = paginated_builds.get_page(paginated_builds.num_pages)
+
+    return render(request, 'ports/all_builds.html', {
+        'all_builds': builds,
+        'filter_applied': filter_applied,
+    })
+
+
+def stats(request):
+    current_week = datetime.datetime.today().isocalendar()[1]
+    all_submissions = User.objects.all()
+    total_unique_users = len(set(all_submissions.values_list('uuid', flat=True)))
+    current_week_unique = len(set(all_submissions.filter(updated_at__week=current_week).values_list('uuid', flat=True)))
```

Comment:
Can you try with `.count()` instead of using `len`?  We only need the count and not the values in memory anyway but I'm not sure if `.count()` works on whatever is returned by `values_list`.

If `values_list` repeats elements and you need unique, try looking for something like `.distinct()`. I'm sure these should be supported as they are pretty common.

Also, I found out you can actually view the SQL query Django executes which may help you debug in some cases. Example: 
`q = User.objects.filter( ... )`
`print(q.query)`

Reply:
> Can you try with `.count()` instead of using `len`? We only need the count and not the values in memory anyway but I'm not sure if `.count()` works on whatever is returned by `values_list`.

Yes, we certainly can avoid going for sets, thank you. `.distinct()` should be the way to go.

> Also, I found out you can actually view the SQL query Django executes which may help you debug in some cases. Example:
> `q = User.objects.filter( ... )`
> `print(q.query)`

Thank you.